src/Nox.Lib/Middlewares/RelatedEndpointsMiddleware.cs

```diff
@@ -0,0 +1,141 @@
+﻿using Microsoft.AspNetCore.Http;
+using Nox.Lib;
+using Nox.Solution;
+
+namespace Nox.Middlewares;
+
+internal class RelatedEndpointsMiddleware
+{
+    private readonly PathString _apiPrefix;
+
+    private readonly RequestDelegate _next;
+
+    private readonly IEnumerable<string> _entitiesPluralNames;
+
+    private readonly IEnumerable<string> _navigationNames;
```

Comment:
Ienumerable hides the fact of doing a query every time/ or not... we should be explicit here

Reply:
updated